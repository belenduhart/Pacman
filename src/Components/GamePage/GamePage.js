import Grid from "./Grid/Grid";
import Report from "./Report/Report";
import * as gp from "../StyledComponents/SC_GridPage";
import { useState } from "react";
import Title from "./Title/Title";
import Buttons from "./Buttons/Buttons";


const GamePage = () => {
  const [showPacman, setShowPacman] = useState(true);
  const [MovementArray, setMovementArray] = useState([{}]);
  const [ShowReport, setShowReport] = useState(false);
  const [output, setOutput] = useState([]);

  window.addEventListener("keydown", PlaceAndReset);

  function PlaceAndReset(Event) {
    if (Event.keyCode === 13) {
      resetGame();
    } else if (Event.keyCode === 82) {
      setShowReport(!ShowReport);
    }
  }
  function resetGame() {
    setMovementArray([]);
    setOutput([]);
    setShowPacman(!showPacman);
    setShowReport(false);
  }

  function ReportHandler(movearray, lastPosition) {
    setMovementArray(movearray);
    setOutput(lastPosition);
    if (output > 0 && movearray.slice(-1) === "MOVE") {
      setOutput(output.slice(-1).split(""));
    }
  }

  return (
    <>
      <Title/>
      <gp.Container>
        <Buttons showPacman={showPacman}/>
        <gp.GridContainer>
          <Grid
            showPacman={!showPacman}
            ReportHandler={ReportHandler}
            resetGame={resetGame}
          />
        </gp.GridContainer>
        <Report output={output} MovementArray={MovementArray} ShowReport={ShowReport}></Report>
      </gp.Container>
    </>
  );
};

export default GamePage;
