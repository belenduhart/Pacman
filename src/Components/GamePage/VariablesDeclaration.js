let UP = 38;
let DOWN = 40;
let LEFT = 37;
let RIGHT = 39;
let MOVE = 32;
let RotateLeft = "img-LEFT";
let RotateRight = "img-RIGHT";
let RotateUp = "img-UP";
let RotateDown = "img-DOWN";
let pacmanClass = "position-zero";


export {UP,DOWN,RIGHT,LEFT,MOVE,RotateDown,RotateLeft,RotateRight,RotateUp,pacmanClass};