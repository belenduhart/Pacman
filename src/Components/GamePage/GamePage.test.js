import GamePage from "./GamePage";
import React from "react";
import Enzyme, { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

it("renders correctly", () => {
  const wrapper = shallow(<GamePage />);
  expect(toJson(wrapper)).toMatchSnapshot();
});


it("Find Pacman after place it", () => {
  const PlaceAndReset = jest.fn();
  const wrapper = shallow(<GamePage onKeyDown={PlaceAndReset} />);
  wrapper.simulate("keydown", { key: "Enter", code: 13, charCode: 13 });
  expect(wrapper.find("div.position-zero")).toBeTruthy();
});



