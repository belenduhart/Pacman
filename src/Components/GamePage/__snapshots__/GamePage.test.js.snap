// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Fragment>
  <Title />
  <styled.div>
    <Buttons
      showPacman={true}
    />
    <styled.div>
      <Grid
        ReportHandler={[Function]}
        resetGame={[Function]}
        showPacman={false}
      />
    </styled.div>
    <Report
      MovementArray={
        Array [
          Object {},
        ]
      }
      ShowReport={false}
      output={Array []}
    />
  </styled.div>
</Fragment>
`;
